### PR TITLE
common: saturate `marginal_feerate()` instead of overflowing

### PR DESCRIPTION
**Pull request by morehouse**
`marginal_feerate()` computed `current_feerate * 1.1` as a double and converted the result back to u32.  Since `current_feerate` is chosen by the peer via `open_channel` or `update_fee`, they could choose an absurdly high value that overflows u32 after the computation.  UBSan reports:

```
common/fee_states.c:179:10: runtime error: 4.72446e+09 is outside the range of representable values of type 'unsigned int'
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior common/fee_states.c:179:10
```

Do the arithmetic with u64 and saturate at `UINT32_MAX` to avoid the undefined behavior.

Found by fuzzing with [smite](https://github.com/morehouse/smite).

### CODE REVIEW

File: common/fee_states.c [hunk 1]
```diff
@@ -175,8 +175,12 @@ u32 marginal_feerate(u32 current_feerate)
 	/* This could happen in future if we celebrate sub-sat summer! */
 	if (current_feerate < minfeerate)
 		current_feerate = minfeerate;
-	if (current_feerate > maxfeerate)
-		return current_feerate * 1.1;
+	if (current_feerate > maxfeerate) {
+		u64 marginal = ((u64)current_feerate * 11) / 10;
```
Comment (Andezion) [inline common/fee_states.c:179]: Is it intentional that the fixed version rounds down ((current_feerate * 11) / 10 truncates), while the old double version also truncated on conversion to u32? I confirmed both truncate the same way for in-range values, so behavior should be unchanged for normal feerates - just confirming this was checked, since it affects the exact fee charged to peers, not just an internal-only value :)

**Review by Andezion — APPROVED** [review]
The fix correctly addresses the UBSan-reported issue, the arithmetic is correct (I verified it numerically), and the added tests cover the exact boundary condition

File: common/fee_states.c (hunk 1 above)
Comment (morehouse) [reply common/fee_states.c:179]: I was simply trying to preserve the existing behavior.  However, I don't think it really matters whether we use floor, round, or ceil here, since this function only influences the ["fee spike buffer"](https://github.com/lightning/bolts/pull/740), whose size is a fuzzy concept to begin with.
